app/src/androidTest/java/ch/hikemate/app/ui/saved/SavedHikesScreenTest.kt

```diff
@@ -360,4 +361,30 @@ class SavedHikesScreenTest : TestCase() {
             .onAllNodesWithTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_HIKE_CARD)
             .assertCountEquals(2)
       }
+
+  @Test
+  fun plannedHikesSectionDisplaysDates() =
+      runTest(timeout = 5.seconds) {
+        val hikes =
+            listOf(
+                detailedHike.copy(
+                    id = "1", name = "Hike 1", plannedDate = Timestamp.now(), isSaved = true),
+                detailedHike.copy(
+                    id = "2", name = "Hike 2", plannedDate = Timestamp.now(), isSaved = true))
+        setupSavedHikes(hikes)
+
+        composeTestRule.setContent { SavedHikesScreen(hikesViewModel, navigationActions) }
+
+        // Select the planned hikes tab
+        composeTestRule
+            .onNodeWithTag(
+                SavedHikesScreen.TEST_TAG_SAVED_HIKES_TABS_MENU_ITEM_PREFIX +
+                    SavedHikesSection.Planned.name)
+            .performClick()
+        composeTestRule.waitForIdle()
+
+        composeTestRule
+            .onAllNodesWithTag(HikeCard.TEST_TAG_IS_SUITABLE_TEXT, useUnmergedTree = true)
```

Comment:
Question: What does `useUnmergedTree` do? Is it mandatory?

Reply:
In my understanding (I must say I'm not sure to know perfectly what it is), `useUnmergedTree = true` tells `onAllNodesWithTag` to look for nested components. Think of it as a UI hierarchy type of thing. In our case there is the `HikeCard` component, and **inside** of it you have the actual message. I want to find the message, so I use `useUnmergedTree`.

What I know 100% is that without it the test does not pass.

Reply:
If the test doesn't pass without it, that's good for me!